collector/src/bin/collector.rs

```diff
@@ -196,7 +196,18 @@ async fn collect_data<'a>(
 
     // Setting up Client
     let grpc_timeout = conf.grpc_timeout;
-    let channel = Channel::from_shared(grpc_host)?.timeout(Duration::from_secs(grpc_timeout));
+
+    // Set up a tls client if required using the locally installed trust store.
+    let endpoint = Channel::from_shared(grpc_host)?.timeout(Duration::from_secs(grpc_timeout));
+    let channel = if endpoint.uri().scheme().map_or(false, |x| x == &http::uri::Scheme::HTTPS) {
+        endpoint.tls_config(ClientTlsConfig::new())?
+    } else {
```

Comment:
I am not sure why this works. By default no TLS roots are enabled. tonic has a feature https://docs.rs/tonic/latest/tonic/#feature-flags `tls-roots` which is not enabled by default.

Perhaps it's enabled by another dependency? But we should be explicit about this regardless.

Reply:
Explicit now.